src/messages/message_accumulator.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::{
+    consensus::{AccumulationError, SignatureAccumulator},
+    messages::{AccumulatingMessage, Message, MessageWithBytes, PlainMessage, SrcAuthority},
+    section::SectionProofChain,
+};
+use serde::{Serialize, Serializer};
+
+#[derive(Default)]
+pub struct MessageAccumulator(SignatureAccumulator<Payload>);
+
+impl MessageAccumulator {
+    pub fn add(&mut self, accumulating_msg: AccumulatingMessage) -> Option<MessageWithBytes> {
+        let payload = Payload {
+            content: accumulating_msg.content,
+            proof_chain: accumulating_msg.proof_chain,
```

Comment:
I am not quite understand why this whole proof_chain need to be part of the accumulating scheme?

Reply:
because It's part of the resulting `Message`.